aten/src/ATen/templates/TensorBody.h

```diff
@@ -204,9 +204,17 @@ class CAFFE2_API Tensor {
     return impl_->is_non_overlapping_and_dense();
   }
 
-  at::MemoryFormat suggest_memory_format() const {
-    if (!is_mkldnn() && !is_sparse() && !impl_->is_contiguous() && impl_->is_strides_like_channels_last()) {
-      return at::MemoryFormat::ChannelsLast;
+  at::MemoryFormat suggest_memory_format(
+      bool channels_last_strides_exact_match = false) const {
+    // Setting channels_last_strides_exact_match to true forces function to
+    // check 0,1 - sized dimention strides.
+    if (!is_mkldnn() && !is_sparse() && impl_->is_strides_like_channels_last()) {
+      if (!channels_last_strides_exact_match && !impl_->is_contiguous()) {
```

Comment:
The logic is strangely aggressively to me. It returns `ChannelsLast` as long as the tensor is not contigous, when we are not requiring exact_match.

Looks like there's an issue here:
Contiguous tensor with size-1 dimension is the case where that flag matters the most. However, if we have a contiguous tensor, `channels_last_strides_exact_match` doesn't really have any effect.